src/main/java/uk/gov/hmcts/ccd/domain/service/processor/FieldProcessorService.java

```diff
@@ -66,6 +81,27 @@ public FieldProcessorService(final List<CaseDataFieldProcessor> caseDataFieldPro
         return processData(data, caseType, eventTriggerService.findCaseEvent(caseType, eventId));
     }
 
+    private CaseViewField processCaseViewField(final CaseViewField field,
+                                               final List<WizardPageField> wizardPageFields) {
+        CaseViewField result = field;
+        for (CaseViewFieldProcessor processor : caseViewFieldProcessors) {
+            result = processor.execute(result, wizardPageField(wizardPageFields, field.getId()));
+        }
+        return result;
+    }
+
+    private List<WizardPageField> getWizardPageFields(String caseTypeId, String eventId) {
+        return uiDefinitionRepository.getWizardPageCollection(caseTypeId, eventId)
+            .stream()
+            .map(WizardPage::getWizardPageFields)
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+    }
+
+    private WizardPageField wizardPageField(List<WizardPageField> wizardPageFields, String fieldId) {
+        return wizardPageFields.stream().filter(f -> f.getCaseFieldId().equals(fieldId)).findAny().orElse(null);
+    }
```

Comment:
minor:
I can see more and more often we need to find a wizard field corresponding to a certain field id. Probably we could introduce a method on the UIDefinitionRepository that returns a map of the wizard fields by fieldID